lib/src/copies.rs

```diff
@@ -121,26 +121,33 @@ impl Stream for CopiesTreeDiffStream<'_> {
     type Item = CopiesTreeDiffEntry;
 
     fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
-        let Some(diff_entry) = ready!(self.inner.as_mut().poll_next(cx)) else {
-            return Poll::Ready(None);
-        };
+        while let Some(diff_entry) = ready!(self.inner.as_mut().poll_next(cx)) {
+            let Some(CopyRecord { source, .. }) = self.copy_records.for_target(&diff_entry.path)
+            else {
+                let target_deleted =
+                    matches!(&diff_entry.value, Ok((_, target_value)) if target_value.is_absent());
+                if target_deleted && self.copy_records.has_source(&diff_entry.path) {
+                    // Skip the "delete" entry when there is a rename.
+                    continue;
```

Comment:
FYI (i.e. not requesting any changes), I was going to make `CopiesTreeDiffEntry` an enum with `Regular`, `Copied`, `Renamed`, and `SourceDeleted` variants. I guess we don't even need the last variant, or not yet at least.

Reply:
I was thinking of a similar change: `Regular(path)|Copied(source, dest)|Renamed(source, dest)`.